app/controllers/ems_common.rb

```diff
@@ -637,9 +637,14 @@ def form_instance_vars
       @server_zones.push([zone.description, zone.name])
     end
     @ems_types = Array(model.supported_types_and_descriptions_hash.invert).sort_by(&:first)
-    @amazon_regions, @azure_regions, @google_regions = %w(Amazon Azure Google).map do |provider_name|
-      get_regions(provider_name)
+
+    @provider_regions = {}
+    model.supported_subclasses.select{|c| c.is_available?(:regions) }.each do |cloud_manager_class|
```

Comment:
A small thing but maybe you *could* leave it in a method, for greater testability :)